code/modules/antagonists/morph/morph.dm

```diff
@@ -75,11 +77,14 @@
 	return !is_type_in_typecache(A, blacklist_typecache) && (isobj(A) || ismob(A))
 
 /mob/living/simple_animal/hostile/morph/proc/eat(atom/movable/A)
+	if(morphed && !eat_while_disguised)
```

Comment:
if it can't eat while disguised and it's disguised fail and tell them why @Shadowflame909